app/src/test/java/pm/gnosis/heimdall/ui/transactiondetails/TransactionDetailsViewModelTest.kt

```diff
@@ -0,0 +1,194 @@
+
+    lateinit var viewModel: TransactionDetailsViewModel
+
+    private val testAddress = "0x0000000000000000000000000000000000000000"
+    private val transactionId = "0000000000000000000000000000000000000000000000000000000000000000"
+    private var confirmTransactionData = "0x${MultiSigWalletWithDailyLimit.ConfirmTransaction.METHOD_ID}$transactionId"
+    private var revokeTransactionData = "0x${MultiSigWalletWithDailyLimit.RevokeConfirmation.METHOD_ID}$transactionId"
+
+    @Before
+    fun setUp() {
+        viewModel = TransactionDetailsViewModel(ethereumJsonRpcRepositoryMock, accountsRepositoryMock, multisigRepositoryMock, gnosisMultisigWrapperMock)
+    }
+
+    @Test
+    fun setConfirmTransaction() {
+        val transaction = TransactionDetails(testAddress.hexAsBigInteger(), data = confirmTransactionData)
+        val testObserver = TestObserver<Unit>()
+
+        viewModel.setTransaction(transaction).subscribe(testObserver)
+
+        assertEquals(BigInteger(transactionId), viewModel.getMultisigTransactionId())
+        assertTrue(viewModel.getMultisigTransactionType() is ConfirmMultisigTransaction)
+        testObserver.assertTerminated().assertNoErrors()
+    }
+
+    @Test
+    fun setRevokeTransaction() {
+        val transaction = TransactionDetails(testAddress.hexAsBigInteger(), data = revokeTransactionData)
+        val testObserver = TestObserver<Unit>()
+
+        viewModel.setTransaction(transaction).subscribe(testObserver)
+
+        assertEquals(BigInteger(transactionId), viewModel.getMultisigTransactionId())
+        assertTrue(viewModel.getMultisigTransactionType() is RevokeMultisigTransaction)
+        testObserver.assertTerminated().assertNoErrors()
+    }
+
+    @Test
+    fun setUnknownTransaction() {
+        val transaction = TransactionDetails(testAddress.hexAsBigInteger(), data = "")
+        val testObserver = TestObserver<Unit>()
+
+        viewModel.setTransaction(transaction).subscribe(testObserver)
+
+        testObserver
+                .assertError { it is IllegalStateException }
+                .assertTerminated()
+    }
+
+    @Test
+    fun setTransactionWithInvalidMultisigAddress() {
+        val transaction = TransactionDetails(BigInteger("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF", 16), data = confirmTransactionData)
+        val testObserver = TestObserver<Unit>()
+
+        viewModel.setTransaction(transaction).subscribe(testObserver)
+
+        testObserver
+                .assertError { it is IllegalStateException }
+                .assertTerminated()
+    }
+
+    @Test
+    fun getMultisigWalletDetails() {
+        val transaction = TransactionDetails(testAddress.hexAsBigInteger(), data = confirmTransactionData)
+        viewModel.setTransaction(transaction).subscribe()
+        given(multisigRepositoryMock.observeMultisigWallet(anyString())).willReturn(Flowable.just(MultisigWallet(testAddress)))
+
+        viewModel.getMultisigWalletDetails()
```

Comment:
Shouldn't we assert the return type here? and test what happens in case of an error, and when events are emitted